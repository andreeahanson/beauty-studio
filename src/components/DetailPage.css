.hello{
  color: red;
  font-size: 100px;
}

.detail-page {
  border: 1px solid blue;
  width: 500px;
  height: 500px;
  color: white
}

.selectedProduct-image {
  height: 200px;
  width: 170px;
}

.x {
  color: red;
}