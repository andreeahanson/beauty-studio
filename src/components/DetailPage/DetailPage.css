.go-back{
  margin-right: 30px; 
  margin-top: 10px;
}

.whole-detail-page {
  border: 1px solid black;
  display: flex;
  flex-direction: row;
  width: 85vw;
  height: 80vh;
  justify-content: center;
  align-items: center;
  padding: 30px;
  overflow: hidden;
  margin-top: 10px; 
}

.detail-page-image{
  width: 40%;
}

.right-side,
.left-side {
  height: 100%;
  width: 50%;
  padding: 20px;
}

li {
  list-style: none;
  text-align: center;
}

.rating{
  height: 5%;
}

ul{
  height: 80px;
  overflow: scroll;
  padding-left: 0px;
}

.link-details{
  height: 10%;;
}

.brand{
  margin: 0px;
  height: 25px;
}

.description{
  height: 150px;
  overflow: scroll;
}

.color-box{
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  grid-gap: 5px;
  overflow: scroll;
  margin: 20px;
  height: 275px;
}




/* .whole-detail-page{
  position: absolute;
  overflow: scroll;
  border: 1px solid red;
  width: 96%;
  display: flex;
}

.align-colors-and-details{
  display: flex;
  flex-direction: column;
  margin-top: 20px;
}

.align {
  border: 1px solid blue;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-content: center;
  font-size: 1em;
}

.detail-page-image {
  height: 29vh;
  width: 22vw;
}

.detail-page {
  width: 40vw;
}

.color-box{
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  grid-gap: 5px;
}

.each-color{
  display: flex;
  flex-direction: column;
}

.align-vertically{
  height: 60px;;
}
 */
