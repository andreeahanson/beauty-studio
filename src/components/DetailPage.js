import React from 'react';
import './DetailPage.css';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';


const DetailPage = ({match, toggleFavorite, id, brand, price, image_link, name, product_type, description, product_colors, favorites}) => {
   let allColors = product_colors.map(color => {
    const divStyle = {
      backgroundColor: color.hex_value
    }
    return (
      <div className="align-vertically">
        <p className="color-names">{color.colour_name}</p>
        <div className="colors" style={divStyle}></div>
      </div>
    )
  })
  return (
    <>
      <figure className="detail-page">
      <h5>{brand}</h5>
      <h3>{name}</h3>
      <h4>$ {price}</h4>
      <img className="selectedProduct-image" src={image_link} alt='beauty-product' />
      <p className='bottom-card'>
      <img onClick={()=>toggleFavorite(id)} className="heart" alt="heart" src={favorites.includes(id) ? 'https://ui-ex.com/images/svg-heart-transparent-background.png' : 'https://images.vexels.com/media/users/3/156965/isolated/preview/cc193f8f5c8e89553153055f410288f9-simple-heart-silhouette-by-vexels.png'} />
      </p>
      <p>{description}</p>
      {allColors}
      </figure>
    </>
  )
}

const mapStateToProps = (state) => ({
  blush: state.blush, 
  bronzer: state.bronzer,
  eyebrow: state.eyebrow,
  eyeliner: state.eyeliner,
  eyeshadow: state.eyeshadow,
  foundation: state.foundation,
  lip_liner: state.lip_liner,
  lipstick: state.lipstick,
  mascara: state.mascara,
  isLoading: state.isLoading,
  favorites: state.favorites
})

export default connect(mapStateToProps)(DetailPage)

DetailPage.propTypes = {
  blush: PropTypes.array,  
  bronzer: PropTypes.array,
  eyebrow: PropTypes.array,
  eyeliner: PropTypes.array,
  eyeshadow: PropTypes.array,
  foundation: PropTypes.array,
  lip_liner: PropTypes.array,
  lipstick: PropTypes.array,
  mascara: PropTypes.array,
  favorites: PropTypes.array,
  match: PropTypes.object,
  toggleFavorite: PropTypes.func
};