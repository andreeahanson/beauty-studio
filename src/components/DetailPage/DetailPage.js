import React from 'react';
import './DetailPage.css';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';


export const DetailPage = ({match, toggleFavorite, id, brand, price, image_link, name, product_type, product_link, description, rating, tag_list, product_colors, favorites}) => {
   let allColors = product_colors.map(color => {
    const divStyle = {
      backgroundColor: color.hex_value
    }
    return (
        <div className="align-vertically">
          <p className="color-names marg">{color.colour_name}</p>
          <div className="colors marg" style={divStyle}></div>
      </div>
    )
  })

    let qualities = tag_list.map(element => <li>{element}</li>)
  return (
    <section className="whole-detail-page">
      <NavLink to={`/${product_type}`} className="go-back detail">Go Back</NavLink>
      <div className="align-colors-and-details">
        <div className="align">
          <img className="detail-page-image" src={image_link} alt='beauty-product' />
          <div className="detail-page">
            <h5>{brand}</h5>
            <h3>{name}</h3>
            <h4>$ {price}</h4>
            <img onClick={()=>toggleFavorite(id)} className="heart" alt="heart" src={favorites.includes(id) ? 'https://ui-ex.com/images/svg-heart-transparent-background.png' : 'https://images.vexels.com/media/users/3/156965/isolated/preview/cc193f8f5c8e89553153055f410288f9-simple-heart-silhouette-by-vexels.png'} />
            <p className="description detail">Description: {description}</p>
            <p className="find-link detail">Find the product here: {product_link}</p>
            <h5 className="rating detail">Rating: {rating ? rating : 'not yet rated'}</h5>
            <h5>Product tags: </h5>
            <ul className="tags detail">{qualities.length ? qualities : 'no tags for this product'}</ul>
          </div>
        </div>
          <h3>Color pallete</h3>
          <div className="color-box">
            {allColors}
          </div>
      </div>
    </section>
  )
}

const mapStateToProps = (state) => ({
  blush: state.blush, 
  bronzer: state.bronzer,
  eyebrow: state.eyebrow,
  eyeliner: state.eyeliner,
  eyeshadow: state.eyeshadow,
  foundation: state.foundation,
  lip_liner: state.lip_liner,
  lipstick: state.lipstick,
  mascara: state.mascara,
  isLoading: state.isLoading,
  favorites: state.favorites
})

export default connect(mapStateToProps)(DetailPage)

DetailPage.propTypes = {
  blush: PropTypes.array,  
  bronzer: PropTypes.array,
  eyebrow: PropTypes.array,
  eyeliner: PropTypes.array,
  eyeshadow: PropTypes.array,
  foundation: PropTypes.array,
  lip_liner: PropTypes.array,
  lipstick: PropTypes.array,
  mascara: PropTypes.array,
  favorites: PropTypes.array,
  match: PropTypes.object,
  toggleFavorite: PropTypes.func
};