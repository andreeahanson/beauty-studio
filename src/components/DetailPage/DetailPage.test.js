import React from 'react';
import { shallow } from 'enzyme';
import { DetailPage, mapStateToProps} from './DetailPage';

describe('DetailPage', () => {
  let wrapper;
  const store = {
    blush: [],
    bronzer: [],
    eyebrow: [],
    eyeliner: [],
    eyeshadow: [],
    foundation: [],
    lip_liner: [],
    lipstick: [],
    mascara: [],
    isLoading: [],
    favorites: []
  }
  let mockToggleFavorite = jest.fn()
  let product_colors = ["red", "yellow"]
  let tag_list = []
  let mockFavorites = [1,2]

  beforeEach(() => {
    wrapper = shallow(
    <DetailPage 
    store={store} 
    product_colors={product_colors} 
    tag_list={tag_list}
    favorites = {mockFavorites}
    />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})