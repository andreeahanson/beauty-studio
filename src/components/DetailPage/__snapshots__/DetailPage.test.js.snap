// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailPage should match the snapshot 1`] = `ShallowWrapper {}`;
